index.bs

```diff
@@ -2540,6 +2542,82 @@ in which case the [=default toJSON operation=] is exposed instead.
 </div>
 
 
+<h4 id=idl-constructors oldids="Constructor" dfn>Constructor methods</h4>
+
+If an [=interface=] has a [=constructor method=] member (matching
+<emu-nt><a href="#prod-Constructor">Constructor</a></emu-nt>), it indicates that it is possible to
+create objects that [=implement=] the [=interface=] using a [=constructor=].
+
+Multiple [=constructor methods=] may appear on a given [=interface=].
+For each [=constructor method=] on the [=interface=], there will be a way to construct an instance
+by passing the specified arguments.
+
+The prose definition of a [=constructor method=] must either initialize the value passed as
+<b>[=this=]</b>, or throw an exception.
+
+A [=constructor method=] must not be defined on a [=callback interface=].
+
+See [[#interface-object]] for details on how a [=constructor method=] is to be implemented.
+
+<div class="example">
+
+    The following IDL defines two interfaces. The second has [=constructor methods=], while the
+    first does not.
+
+    <pre highlight="webidl">
+        [Exposed=Window]
+        interface NodeList {
+          Node item(unsigned long index);
+          readonly attribute unsigned long length;
+        };
+
+        [Exposed=Window]
+        interface Circle {
+          constructor();
+          constructor(double radius);
+          attribute double r;
+          attribute double cx;
+          attribute double cy;
+          readonly attribute double circumference;
+        };
+    </pre>
+
+    An ECMAScript implementation supporting these interfaces would have a \[[Construct]] property
+    on the <code class="idl">Circle</code> interface object which would return a new object that
+    [=implements=] the interface.
+    It would take either zero or one argument.
+    The <code class="idl">NodeList</code> interface object would not have a \[[Construct]]
+    property.
```

Comment:
I thought it would have a throwing one? And the way we'd explain how an implementation was able to create it was by passing some magical token.

Reply:
Unclear: https://github.com/heycam/webidl/issues/698.

Reply:
We should either have this be an XXX unclear box, or block this on resolving #698.